WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsV4PeriodViewController.swift

```diff
@@ -614,9 +617,27 @@ private extension StoreStatsV4PeriodViewController {
 
     func updateOrderDataIfNeeded() {
         orderStatsIntervals = orderStats?.intervals.sorted(by: { (lhs, rhs) -> Bool in
-            return lhs.dateStart() < rhs.dateStart()
+            guard let lhsDateStart = lhs.dateStart(), let rhsDateStart = rhs.dateStart() else {
+                return true
+            }
+            return lhsDateStart < rhsDateStart
         }) ?? []
-        if let startDate = orderStatsIntervals.first?.dateStart(),
+
+        guard let firstStatsInterval = orderStatsIntervals.first else {
+            return
+        }
+
+        guard firstStatsInterval.dateStart() != nil, firstStatsInterval.dateEnd() != nil else {
+            let errorInfo = "Failed to parse stats v4 interval date(s)"
+            let error = NSError(domain: "WooCommerceStatsV4ErrorDomain",
+                                code: 100,
+                                userInfo: [NSLocalizedDescriptionKey: errorInfo])
+            let errorMessage = "Failed to parse stats interval start date: \(firstStatsInterval.dateStart); end date: \(firstStatsInterval.dateEnd)"
+            CrashLogging.logError(error, userInfo: ["Message": errorMessage], level: .error)
```

Comment:
If you like, you could use `CrashLogging.logMessage` here to avoid the complexity of putting together an `NSError`. Up to you though! 😃